ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java

```diff
@@ -52,59 +46,60 @@ public Boolean supportsExecutionPayload() {
   }
 
   @Override
-  public BeaconBlockBodyBuilder executionPayload(SafeFuture<ExecutionPayload> executionPayload) {
-    if (!isBlinded()) {
-      this.executionPayload = executionPayload;
-    }
+  public BeaconBlockBodyBuilder executionPayload(
+      final SafeFuture<ExecutionPayload> executionPayload) {
+    this.executionPayload = executionPayload;
     return this;
   }
 
   @Override
   public BeaconBlockBodyBuilder executionPayloadHeader(
-      SafeFuture<ExecutionPayloadHeader> executionPayloadHeader) {
-    if (isBlinded()) {
-      this.executionPayloadHeader = executionPayloadHeader;
-    }
+      final SafeFuture<ExecutionPayloadHeader> executionPayloadHeader) {
+    this.executionPayloadHeader = executionPayloadHeader;
     return this;
   }
 
   @Override
-  protected void validateSchema() {
-    if (isBlinded()) {
-      checkState(
-          blindedSchema != null && schema == null, "blindedSchema must be set with no schema");
+  @SuppressWarnings("unchecked")
+  protected <T> T getAndValidateSchema(final boolean blinded, final Class<T> expectedSchemaType) {
+    final BeaconBlockBodySchema<?> schema;
+    if (blinded) {
+      schema = this.blindedSchema;
```

Comment:
`this.` is a bit overkill here and 3 lines down, I think it could be safely removed